lib/neo4j/active_node/spatial.rb

```diff
@@ -5,38 +5,75 @@ def self.included(other)
         other.extend(ClassMethods)
       end
 
-      def add_to_spatial_index(index_name = nil)
-        index = index_name || self.class.spatial_index_name
-        fail 'index name not found' unless index
-        Neo4j::Session.current.add_node_to_spatial_index(index, self)
+      def add_to_spatial_layer(layer_name = nil)
+        layer = layer_name || self.class.spatial_layer_name
+        fail 'layer name not found' unless layer
+        Neo4j::ActiveBase.current_session.add_node_to_layer(layer, self)
       end
 
       module ClassMethods
-        attr_reader :spatial_index_name
-        def spatial_index(index_name = nil)
-          return spatial_index_name unless index_name
-          # create_index_callback(index_name)
-          @spatial_index_name = index_name
+        attr_reader :spatial_layer_name
+        attr_reader :spatial_layer_type
+        attr_reader :spatial_layer_config
+
+        def spatial_layer(layer_name = nil, options = {})
+          return spatial_layer_name unless layer_name
+
+          @spatial_layer_name = layer_name
+          @spatial_layer_type = options.fetch(:type, 'SimplePoint')
+          @spatial_layer_config = options.fetch(:config, 'lon:lat')
+
+          spatial_layer_name
         end
 
-        # This will not work for now. Neo4j Spatial's REST API doesn't seem to work within transactions.
-        # def create_index_callback(index_name)
-        #   after_create(proc { |node| Neo4j::Session.current.add_node_to_spatial_index(index_name, node) })
-        # end
+        def create_layer
+          fail 'layer name not found' unless spatial_layer_name
 
-        # private :create_index_callback
+          lon_name, lat_name = spatial_layer_config.split(':')
+
+          Neo4j::ActiveBase.current_session.add_layer(spatial_layer_name, spatial_layer_type, lat_name, lon_name)
```

Comment:
This might just be part of the original API, but it feels strange to extend the session objects with `add_layer` / `remove_layer` / `within_distance` / etc...  I would expect maybe some other object which is given a session on initialization.  Depends on if you find that worthwhile and if you want to go to the extent of changing it